### ISSUE TITLE
Email Times Out without any logging

### ISSUE THREAD

**Issue by bakerboy448** [janeczku/calibre-web#2179]
**Describe the bug/problem**
A clear and concise description of what the bug is. If you are asking for support, please check our [Wiki](https://github.com/janeczku/calibre-web/wiki) if your question is already answered there.

Unable to test email and minimal logging means it cannot easily be debugged either

**To Reproduce**
1. Add email account with known working credentials and password.
2. Test Email Fails
3. 
**Logfile**
Add content of calibre-web.log file or the relevant error, try to reproduce your problem with "debug" log-level to get more output.

```
[2021-11-14 16:11:48,366]  INFO {cps:140} Starting Calibre Web...
[2021-11-14 16:11:49,055]  INFO {cps.server:204} Starting Gevent server on [::]:8083
[2021-11-14 18:28:10,446] DEBUG {cps.updater:326} Nightly version: 974549b1af7addc84b8012926f03188ec74b625d, 2021-11-08T20:36:41+04:00
[2021-11-14 18:40:43,982]  INFO {cps.server:272} webserver stop (restart=False)
[2021-11-14 18:40:43,983]  INFO {cps.server:257} Performing shutdown of Calibre-Web
[2021-11-14 18:40:45,441] DEBUG {cps.services:35} Cannot import simpleldap, logging in with ldap will not work: No module named 'flask_simpleldap'
[2021-11-14 18:40:46,323]  INFO {cps:140} Starting Calibre Web...
[2021-11-14 18:40:46,401]  INFO {cps.server:204} Starting Gevent server on [::]:8083
[2021-11-14 18:44:47,686] DEBUG {cps.updater:326} Nightly version: 974549b1af7addc84b8012926f03188ec74b625d, 2021-11-08T20:36:41+04:00
[2021-11-14 18:45:14,174] DEBUG {cps.config_sql:337} _ConfigSQL updating storage
[2021-11-14 18:45:15,382] DEBUG {cps.services.worker:88} Add Task for user: bakerboy448: E-mail, Calibre-Web test e-mail
[2021-11-14 18:45:15,386] DEBUG {cps.tasks.mail:186} Start sending e-mail
[2021-11-14 18:45:23,085] DEBUG {cps.updater:326} Nightly version: 974549b1af7addc84b8012926f03188ec74b625d, 2021-11-08T20:36:41+04:00
[2021-11-14 18:46:19,945] ERROR {cps.tasks.mail:175} [Errno 110] Connection timed out
```

**Expected behavior**
Emails should be sendable

**Screenshots**

**Environment (please complete the following information):**
 - OS: [e.g. Windows 10/Raspberry Pi OS]  Ubuntu 20.04.3 LTS
 - Python version: [e.g. python2.7] Python 3.8.10
 - Calibre-Web version: [e.g. 0.6.8 or 087c4c59 (git rev-parse --short HEAD)]: 974549b1af7addc84b8012926f03188ec74b625d
 - Docker container: [None/LinuxServer]: None
 - Special Hardware: [e.g. Rasperry Pi Zero] N/A
 - Browser: [e.g. Chrome 83.0.4103.97, Safari 13.3.7, Firefox 68.0.1 ESR] Agnostic

**Additional context**
Add any other context about the problem here. [e.g. access via reverse proxy, database background sync, special database location]

**Comment by wgibcus** [comment]
Can you provide details of the mail service you are using?

**Comment by bakerboy448** [comment]
Gmail regular SMTP tried both 465 and 587 with app password that is known to work in other cases.  Only CalibreWeb is having issues sending emails from the same server at this point.

it did work before, but stopped recently with one of the updates - not sure which sadly.

**Comment by wgibcus** [comment]
It would be useful to include the server settings and port numbers etc. in your request (minus your email address) to further this request. Email details for Gmail should be:
SMTP server: smtp.gmail.com
Port: 465
Encryption: SSL/TLS
Login: fulladdress@gmail.com
From email: sameaslogin@gmail.com

Make sure your firewall is configured to allow traffic through port 465.

**Comment by bakerboy448** [comment]
- have tried several combinations (see below)
- it's not a firewall issue
- Readarr/Radarr/Lidarr/Sonarr/Prowlarr/Notifiarr/Webmin all have 0 issues using the same settings (even the same app password)

SMTP Hostname: smtp.google.com
SMTP Port: 587 and 465
Encryption: STARTTLS and SSL/TLS
SMTP Login: myusername and myusername@gmail.com
From E-mail: myusername@gmail.com

All result in a timeout and effectively nothing logged even in debug

**Comment by OzzieIsaacs** [comment]
Well `{cps.tasks.mail:175} [Errno 110] Connection timed out` is not nothing logged. 
This is the error message, so calibre-web can't establish an connection to the host for whatever reason. 
Calibre-web also supports the "native/oauth" googlemail authentification. For setting this up, please refer to https://github.com/janeczku/calibre-web/wiki/Setup-Mailserver#gmail (not so complicated as it might look on the first sight)

**Comment by bakerboy448** [comment]
Well calibre web is the only application on the entire server that magically timed out to gmail's SMTP server. Were there any change sim recent beta versions? This used to work before without issue. Can you add additional logging around the connection it's attempting or let me know where it's at and I can patch it in manually?

**Comment by bakerboy448** [comment]
Attempting to setup Oauth in Calibreweb ends up with a 504 and calibrweb is not responsive.
trying to access via IP:port directly times out.  Logs show literally nothing at all.

I had restarted; and then tried to use the OAUTH, hitting configure just results in calibre web being completely unresponsive
```
[2021-11-20 08:45:50,679]  INFO {cps.server:272} webserver stop (restart=False)
[2021-11-20 08:47:27,353] DEBUG {cps.services:35} Cannot import simpleldap, logging in with ldap will not work: No module named 'flask_simpleldap'
[2021-11-20 08:47:32,376]  INFO {cps:140} Starting Calibre Web...
[2021-11-20 08:47:32,506]  INFO {cps.server:204} Starting Gevent server on [::]:8083
```

gmail.json is in the calibreweb directory and owned by the user/group calibreweb is running as with 644 permissions.

**Comment by OzzieIsaacs** [comment]
While doing the oauth stuff calibre-web (or the underlying library) is opening a random port and contact google server, looks like this also waiting for a timeout and is therefore unresponsive. Still looks to me like there is a firewall, or something else in between preventing the connection to happen.

Recent changes in email itself nothing relevant I think (removed code for python 2 and renamed mail/e-mail/email texts):
![grafik](https://user-images.githubusercontent.com/6438895/142732324-2a19586d-802b-4c78-b526-b1aaa412acd9.png)

The error message has some more information to show the number "110". A short google search turns out: https://stackoverflow.com/questions/45253124/linux-socket-recv-return-1-errno-110-connection-timed-out
The error message is created based on a socket error in calibre-web (low level connection)

Regarding additional logging to check.
Just add lines like `log.debug("random text")` to the functions in mail.py (where you've found the 5min timeout). If there is nothing logged, than my bet is the first connection attempt in line 188 fails. If the conection is successfully established, you should see the raw email traffic like:
```
[2021-10-30 21:28:07,348] DEBUG {cps.tasks.mail:68} send: ehlo [127.0.1.1]
[2021-10-30 21:28:07,351] DEBUG {cps.tasks.mail:66} send: AUTH PLAIN
[2021-10-30 21:28:07,356] DEBUG {cps.tasks.mail:68} send: mail FROM:<name@host.com> size=534
[2021-10-30 21:28:07,362] DEBUG {cps.tasks.mail:68} send: rcpt TO:<a5@b.com>
[2021-10-30 21:28:07,364] DEBUG {cps.tasks.mail:68} send: data
[2021-10-30 21:28:07,367] DEBUG {cps.tasks.mail:68} send: b'Content-Type: multipart/mixed; boundary="===============7764683185662062346=="\r\nMIME-Version: 1.0\r\nto: a5@b.com\r\nfrom: name@host.com\r\nsubject: Calibre-Web test e-mail\r\nMessage-Id: \r\n <163562208732.104867.17212347788275773249.calibre-web@mint-desktop>\r\nDate: Sat, 30 Oct 2021 21:28:07 +0200\r\n\r\n--====='
[2021-10-30 21:28:07,376] DEBUG {cps.tasks.mail:68} send: quit
[2021-10-30 21:28:07,380] DEBUG {cps.tasks.mail:209} E-mail send successfully
```
Are you using ipv6 or ipv4?

**Comment by OzzieIsaacs** [comment]
If you are having debug logging enable there should be a traceback visible after the error message like:
```
Traceback (most recent call last):
  File "/home/ozzie/Development/calibre-web/cps/tasks/mail.py", line 157, in run
    self.send_standard_email(msg)
  File "/home/ozzie/Development/calibre-web/cps/tasks/mail.py", line 191, in send_standard_email
    self.asyncSMTP = Email(self.settings["mail_server"], self.settings["mail_port"], timeout=timeout)
  File "/home/ozzie/Development/calibre-web/cps/tasks/mail.py", line 111, in __init__
    smtplib.SMTP.__init__(self, *args, **kwargs)
  File "/usr/lib/python3.8/smtplib.py", line 255, in __init__
    (code, msg) = self.connect(host, port)
  File "/usr/lib/python3.8/smtplib.py", line 339, in connect
    self.sock = self._get_socket(host, port, self.timeout)
  File "/usr/lib/python3.8/smtplib.py", line 310, in _get_socket
    return socket.create_connection((host, port), timeout,
  File "/usr/lib/python3.8/socket.py", line 787, in create_connection
    for res in getaddrinfo(host, port, 0, SOCK_STREAM):
  File "/usr/lib/python3.8/socket.py", line 918, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
socket.gaierror: [Errno -2] Name or service not known
```
Which should also give some more information

**Comment by OzzieIsaacs** [comment]
I checked with my computer with a generic block port 465 rule I get network is unreachable after 2min10sec
```
[2021-11-20 17:00:17,164] ERROR {cps.tasks.mail:48} [Errno 101] Network is unreachable
Traceback (most recent call last):
  File "/home/ozzie/Development/calibre-web/cps/tasks/mail.py", line 157, in run
    self.send_standard_email(msg)
  File "/home/ozzie/Development/calibre-web/cps/tasks/mail.py", line 188, in send_standard_email
    self.asyncSMTP = EmailSSL(self.settings["mail_server"], self.settings["mail_port"],
  File "/home/ozzie/Development/calibre-web/cps/tasks/mail.py", line 118, in __init__
    smtplib.SMTP_SSL.__init__(self, *args, **kwargs)
  File "/usr/lib/python3.8/smtplib.py", line 1043, in __init__
    SMTP.__init__(self, host, port, local_hostname, timeout,
  File "/usr/lib/python3.8/smtplib.py", line 255, in __init__
    (code, msg) = self.connect(host, port)
  File "/usr/lib/python3.8/smtplib.py", line 339, in connect
    self.sock = self._get_socket(host, port, self.timeout)
  File "/usr/lib/python3.8/smtplib.py", line 1049, in _get_socket
    new_socket = socket.create_connection((host, port), timeout,
  File "/usr/lib/python3.8/socket.py", line 808, in create_connection
    raise err
  File "/usr/lib/python3.8/socket.py", line 796, in create_connection
    sock.connect(sa)
OSError: [Errno 101] Network is unreachable
```
Looks like some of the system libraries having shorter timeouts than 5min, (2min for me), and a blocking firewall gives another error message, but I see the traceback (Linux Mint) and I can continue afterwards (send the next failing email)

**Comment by bakerboy448** [comment]
(just testing with smtp at this point as oauth makes calibreweb completely unresponsive and literally nothing is logged there. browser dev console shows that it was still using the smtp form info)

disabled UFW  to remove the firewall from the equation - still timed out at 1:04

debug is enabled and logs from calibre-web.log show nothing at all other than what I've already shared

next disabled ipv6, restarted the server, re-disabled ufw, tried to test again:

```
bakerboy448@hetz-baker-2004:~$ sysstatus calibreweb
● calibreweb.service - Calibre-Web
     Loaded: loaded (/etc/systemd/system/calibreweb.service; enabled; vendor preset: enabled)
     Active: active (running) since Sat 2021-11-20 10:10:38 CST; 7min ago
```

```
[2021-11-20 10:11:06,228] DEBUG {cps.services:35} Cannot import simpleldap, logging in with ldap will not work: No module named 'flask_simpleldap'
[2021-11-20 10:11:17,673]  WARN {fake_useragent:180} Error occurred during loading data. Trying to use cache server https://fake-useragent.herokuapp.com/browsers/0.1.11
Traceback (most recent call last):
  File "/opt/.venv/calibreweb/lib/python3.8/site-packages/fake_useragent/utils.py", line 154, in load
    for item in get_browsers(verify_ssl=verify_ssl):
  File "/opt/.venv/calibreweb/lib/python3.8/site-packages/fake_useragent/utils.py", line 99, in get_browsers
    html = html.split('<table class="w3-table-all notranslate">')[1]
IndexError: list index out of range
[2021-11-20 10:11:18,428]  INFO {cps:140} Starting Calibre Web...
[2021-11-20 10:11:18,783]  INFO {cps.server:204} Starting Gevent server on [::]:8083
[2021-11-20 10:11:45,062] DEBUG {cps.updater:326} Nightly version: 974549b1af7addc84b8012926f03188ec74b625d, 2021-11-08T20:36:41+04:00
[2021-11-20 10:15:16,849] DEBUG {cps.config_sql:337} _ConfigSQL updating storage
[2021-11-20 10:15:16,854] DEBUG {cps.services.worker:88} Add Task for user: bakerboy448: E-mail, Calibre-Web test e-mail
[2021-11-20 10:15:16,861] DEBUG {cps.tasks.mail:186} Start sending e-mail
[2021-11-20 10:16:21,437] ERROR {cps.tasks.mail:175} [Errno 110] Connection timed out
```

**Comment by OzzieIsaacs** [comment]
What's this?
```
Error occurred during loading data. Trying to use cache server https://fake-useragent.herokuapp.com/browsers/0.1.11
Traceback (most recent call last):
  File "/opt/.venv/calibreweb/lib/python3.8/site-packages/fake_useragent/utils.py", line 154, in load
    for item in get_browsers(verify_ssl=verify_ssl):
  File "/opt/.venv/calibreweb/lib/python3.8/site-packages/fake_useragent/utils.py", line 99, in get_browsers
    html = html.split('<table class="w3-table-all notranslate">')[1]
IndexError: list index out of range
```

**Comment by bakerboy448** [comment]
Your guess is as good as mine for that.  In the 2 other archived log files since 11/11 - that error only appears 1 other time, but appears to be related (and issues in other apps after this reboot confirmed similar issues) just a few seconds after reboot of network gremlins (resource unavailable for those apps)

but that does not reoccur any other time

**Comment by OzzieIsaacs** [comment]
Runing out of ideas, could you please try to update the dependencies
`[wherever]/python3 -m pip install --upgrade -r [wherever]/requirements.txt`

**Comment by OzzieIsaacs** [comment]
I just doublechecked the ouath gmail stuff, for me it's working. 
Outputting a lot of technospeak
![grafik](https://user-images.githubusercontent.com/6438895/142734033-34e26dcc-f61d-4b01-b90c-fa2bb96bcfc7.png)

**Comment by bakerboy448** [comment]
updated deps https://gist.github.com/bakerboy448/b382e2d77e1958052e4160f640bd91ed
then restarted calibre web and tried again; still nothing
```
[2021-11-20 10:36:58,754]  INFO {cps.server:272} webserver stop (restart=False)
[2021-11-20 10:36:58,754]  INFO {cps.server:257} Performing shutdown of Calibre-Web
[2021-11-20 10:37:00,912] DEBUG {cps.services:35} Cannot import simpleldap, logging in with ldap will not work: No module named 'flask_simpleldap'
[2021-11-20 10:37:01,741]  INFO {cps:140} Starting Calibre Web...
[2021-11-20 10:37:01,826]  INFO {cps.server:204} Starting Gevent server on [::]:8083
[2021-11-20 10:37:23,004] DEBUG {cps.updater:326} Nightly version: 974549b1af7addc84b8012926f03188ec74b625d, 2021-11-08T20:36:41+04:00
[2021-11-20 10:37:36,404] DEBUG {cps.config_sql:337} _ConfigSQL updating storage
[2021-11-20 10:37:36,754] DEBUG {cps.services.worker:88} Add Task for user: bakerboy448: E-mail, Calibre-Web test e-mail
[2021-11-20 10:37:36,758] DEBUG {cps.tasks.mail:186} Start sending e-mail
[2021-11-20 10:38:41,341] ERROR {cps.tasks.mail:175} [Errno 110] Connection timed out
```

for oauth - the issue is I can't even attempt to configure it as clicking the setup button causes calibreweb to hang and eventually timeout with no indication of anything occurring in the logs

**Comment by OzzieIsaacs** [comment]
If you click on the configure button, it should open a new browser windows showing the "!select your account" screen: 
![grafik](https://user-images.githubusercontent.com/6438895/142734308-e609d086-216b-4862-91af-2c7b39e2388f.png)
It's indeed if you close this windows (or it doesn't show up) the game in calibre-web is over, it's showing the moving loading dot (firefox) in the tab. I check if I can change this, but this will not help you. 

I have no more ideas why it's not working for you.

You could try to revert to the working version and move step by step forward till it breaks. Maybe we than can figure out what's wrong.

**Comment by bakerboy448** [comment]
> It's indeed if you close this windows (or it doesn't show up) the game in calibre-web is over, it's showing the moving loading dot (firefox) in the tab. I check if I can change this, but this will not help you.

Yup it never shows; if it makes a difference at all - I am going through a nginx reverse proxy 

ended up manually editing the database to remove the email settings...which seemed to cause a partial reset of the database.
it reverted back to the 11/8 build and dropped the settings (kept the users though)
updated to nightly 11/20 and reconfigured gmail using the same settings as before and now it works again....weird

**Comment by delize** [comment]
I believe I might be seeing this same thing (or something similar) happen in a Docker container.

I have the gmail.json file uploaded to the proper location, and as soon as I hit the Authenticate button, I get a 504 a few minutes later. If I try to navigate to any other bit of the interface, it results in 504. The only solution so far I have found is to restart the docker container.

I bumped up the log file to Debug before doing this, but, it isn't writing any debug log context. 

In this case however, I am using a. linuxserver docker container against their `nightly` build.
